src/AzureAppConfigurationImpl.ts

```diff
@@ -514,30 +524,61 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             return false;
         };
 
-        const needRefresh: boolean = await this.#executeWithFailoverPolicy(funcToExecute);
-        if (needRefresh) {
-            await this.#loadFeatureFlags();
-        }
-
-        this.#featureFlagRefreshTimer.reset();
-        return Promise.resolve(needRefresh);
+        const isChanged = await this.#executeWithFailoverPolicy(funcToExecute);
+        return isChanged;
     }
 
-    onRefresh(listener: () => any, thisArg?: any): Disposable {
-        if (!this.#refreshEnabled && !this.#featureFlagRefreshEnabled) {
-            throw new Error("Refresh is not enabled for key-values or feature flags.");
+    /**
+     * Gets a configuration setting by key and label.If the setting is not found, return undefine instead of throwing an error.
+     */
+    async #getConfigurationSetting(configurationSettingId: ConfigurationSettingId, customOptions?: GetConfigurationSettingOptions): Promise<GetConfigurationSettingResponse | undefined> {
+        const funcToExecute = async (client) => {
+            return getConfigurationSettingWithTrace(
+                this.#requestTraceOptions,
+                client,
+                configurationSettingId,
+                customOptions
+            );
+        };
+
+        let response: GetConfigurationSettingResponse | undefined;
+        try {
+            response = await this.#executeWithFailoverPolicy(funcToExecute);
+        } catch (error) {
+            if (isRestError(error) && error.statusCode === 404) {
+                response = undefined;

```

Comment:
When customer enabled keyValue refresh and specified a non-existing sentinel, in this scenario it will never trigger a refresh right? Should we throw error or log warning to let the customer know the non-existing sentinel?

Reply:
When the sentinel key is created, it will trigger a refresh. 404 is not an error case.